common/telemetry.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * This file is part of Bott.
+ *
+ * This project is dual-licensed:
+ * - Non-commercial use: AGPLv3 (see LICENSE file for full text).
+ * - Commercial use: Proprietary License (contact D@nielLaCos.se for details).
+ *
+ * Copyright (C) 2025 DanielLaCos.se
+ */
+
+import { type Span, trace, type Tracer } from "@opentelemetry/api";
+
+const TRACER_NAME = "bott";
+
+export function getTracer(): Tracer {
+  return trace.getTracer(TRACER_NAME);
+}
+
+export function startSpan(
+  name: string,
+  attributes?: Record<string, unknown>,
+): Span {
+  const tracer = getTracer();
+  const span = tracer.startSpan(name);
+
+  if (attributes) {
+    for (const [key, value] of Object.entries(attributes)) {
+      if (value !== undefined && value !== null) {
+        span.setAttribute(key, String(value));
+      }
+    }
+  }
+
+  return span;
+}
+
+export function withSpan<T>(
```

Comment:
eh, I don't love the `withSpan` and `withSpanAsync` - makes the code more nested. let's just use "createSpan"